topics/w6-live-programming.md

```diff
@@ -1,11 +1,70 @@
 # W6 - Live Programming
+## SnipPy: Small-step live programming by example
 
-### Reading List
+SnipPy is a tool that combines two user assisting programming paradigms: live programming and programming by example.
+Live programming refers to programming environments that provide near-instant feedback, enabling programmers to immediately observe the effects of code changes. This can range from simple tools like REPLs (Read-Eval-Print Loops) to systems like PythonTutor (Week 4) or Sketch-n-sketch (Week 5), which offer visualizations and interactive debugging. On the other hand, Programming by example focuses on generating code by taking input/output pairs as demonstrations. A synthesizer analyzes these pairs and attempts to infer a program that produces the desired results
 
-#### Required Reading
+In SnipPy, live programming is facilitated by projection boxes, which dynamically display the current state of program variables beside each line of code. Programmers can step through the code line by line and observe how the program state changes in real time. This visualization is a powerful aid in writing code, debugging, and understanding new code, as it allows developers to reason visually about how their code manipulates data. (INSERT EXAMPLE IMAGE FOR PROJECTION BOXES)
 
+Building upon this snippy now integrates programming by example into projection boxes. On a new line of code, the user can provide input/output pairs into a projection box and the tool will try to synthesize a matching expression.
+
+The inner workings of the synthesizer are quite easy to understand. It has an abstract grammar that 
```

Comment:
The focus on the synthesizer feels overemphasized and takes attention away from the main topic of live programming. 
Perhaps realign the discussion, making the synthesizer a secondary element that enables the design of live programming.